run_in_parallel.py

```diff
@@ -0,0 +1,242 @@
+        return False
+
+    is_ignore_sequential_data_setup = strtobool(os.getenv('IGNORE_SEQUENTIAL_DATA_SETUP'))
+
+    return is_ignore_sequential_data_setup
+
+
+def parse_arguments():
+    """
+    Parses commandline arguments
+    :return: Parsed arguments
+    """
+    parser = argparse.ArgumentParser('Run behave in parallel mode for scenarios')
+    parser.add_argument('--command_line_args', '-a', help='Command line arguments', default=DEFAULT_COMMAND_LINE_ARGS)
+    parser.add_argument('--format', '-f', help='Behave format', default=DEFAULT_BEHAVE_FORMAT)
+    parser.add_argument('--acceptance_features_directory', '-d', help='specify directory containing features',
+                        default=DEFAULT_FEATURES_DIRECTORY)
+    parser.add_argument('--processes', '-p', type=int,
+                        help=f'Maximum number of processes. Default [{DEFAULT_PROCESSES}] ', default=DEFAULT_PROCESSES)
+    parser.add_argument('--tags', '-t', help='specify behave tags to run', default=DEFAULT_TAGS)
+    parser.add_argument('--timeout', '-tout', type=int,
+                        help='Maximum seconds to execute each scenario. Default = 300', default=300)
+
+    # TODO BUG: "AND" tags not handled properly
+    args = parser.parse_args()
+
+    return args
+
+
+def is_process_running(process):
+    return process != None and process.is_alive()
+
+
+def _run_scenario(q, feature_scenario, timeout, command_line_args):
+    """
+    Runs features/scenarios
+    :param feature_scenario: Feature/scenario that should be run
+    :type feature_scenario: str
+    :return: Feature/scenario and status
+    """
+
+    execution_code = {0: 'OK', 1: 'FAILED', 2: 'TIMEOUT'}
+    feature, scenario = feature_scenario.split(DELIMITER)
+
+    cmd = f'behave {command_line_args} --format progress2 {feature} --name \"{scenario}\"'
+
+    try:
+        check_output(cmd, shell=True, timeout=timeout)
+        code = 0
+    except CalledProcessError as e:
+        out_bytes = e.output
+        code = e.returncode
+    except TimeoutExpired:
+        code = 2
+
+    status = execution_code[code]
+    logger.info(f"{feature:50}: {scenario} --> {status}")
+
+    if status == 'FAILED':
+        q.put(f'FAILED - Feature: [{feature}], Scenario [{scenario}]"')
+        logger.error(out_bytes.decode())
+
+    time.sleep(10)
+
+    return feature, scenario, status
+
+
+def run_all_scenarios(scenarios_to_run, thread_count, timeout, command_line_args):
+    total_scenarios_to_run = len(scenarios_to_run)
+
+    # Set number of threads needed
+    if total_scenarios_to_run < thread_count:
+        process_pool_size = total_scenarios_to_run
+    else:
+        process_pool_size = thread_count
+
+    process_pool = [None] * process_pool_size
+    scenario_index = 0
+    processes_running = True
+    failure_queue = Queue(maxsize=0)
+
+    # Run every scenario
+    while processes_running:
+
+        # Find a 'Free' Thread slot
+        for process_index in range(process_pool_size):
```

Comment:
You could iterate directly over `process_pool` rather than using indexes.

Reply:
changed the 2nd loop to use  "for process in process_pool", how would I use that in the other as  I need to replace the process with new ones?